### PR TITLE
*: change dir named with etcd-restore-operator to restore-operator

### PR DESCRIPTION
**Pull request by fanminshi**
change all dir named with etcd-restore-operator to restore-operator.

### CODE REVIEW

**Comment by fanminshi** [comment]
cc/ @hongchaodeng @hasbro17

File: hack/build/restoreoperator/build [hunk 1]
```diff
@@ -6,7 +6,7 @@ set -o pipefail
 
 function go_build {
 	echo "building "${1}"..."
-	GOOS=linux GOARCH=amd64 CGO_ENABLED=0 go build -o ${bin_dir}/${1} -installsuffix cgo -ldflags "$go_ldflags" ./cmd/${1}/
+	GOOS=darwin GOARCH=amd64 CGO_ENABLED=0 go build -o ${bin_dir}/etcd-restore-operator -installsuffix cgo -ldflags "$go_ldflags" ./cmd/${1}/
```
Comment (hongchaodeng) [inline hack/build/restoreoperator/build]: Should be `GOOS=linux` ?

File: cmd/restoreoperator/main.go [hunk 2]
```diff
@@ -22,7 +22,7 @@ import (
 
 	"github.com/coreos/etcd-operator/pkg/controller/restorecontroller"
 	"github.com/coreos/etcd-operator/pkg/util/k8sutil"
-	version "github.com/coreos/etcd-operator/version/etcd-restore-operator"
+	version "github.com/coreos/etcd-operator/version/restoreoperator"
```
Comment (xiang90) [inline cmd/restoreoperator/main.go]: restore-operator is probably cleaner.

File: hack/build/restoreoperator/build (hunk 1 above)
Comment (fanminshi) [reply hack/build/restoreoperator/build]: good catch.

File: cmd/restoreoperator/main.go (hunk 2 above)
Comment (hongchaodeng) [reply cmd/restoreoperator/main.go]: So some dirs are "restoreoperator" but some are "restore-operator".
Isn't it better to unify them?
I have no preference on either.

**Comment by fanminshi** [comment]
all fixed. PTAL

**Comment by hongchaodeng** [comment]
lgtm

**Pull request merged**